Documentation/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,56 @@
+# Ceph bucket notifications CRD
+
+## Overview
+
+In its new release, Ceph introduced the bucket notifications feature. It allows to send messages to various endpoints when a new event occurs on a bucket (ref: <https://docs.ceph.com/docs/master/radosgw/notifications/)>
+
+Setup of those notifications are normally done by sending HTTP requests to the RGW, either to create/delete topics pointing to specific endpoints, or create/delete bucket notifications based on those topics.
+
+This functionality eases this process by avoiding to use external tools or scripts. It is replaced by creation of CR definitions that  contain all the information necessary to create topics and/or notifications, which the rook operator processes.
+
+## Goals
+
+Creates a CRD for topics and a CRD for notifications, defining all the necessary and optional information for the various endpoints.
+
+Extends the rook operator to handle the CRs that would be submitted by users.
+
+## Implementation
+
+The CR for a topic configuration takes this form:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: Topic
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+Spec:
+  pushEndpoint: Kafka/HTTP/RabbitMQ, mandatory
+  ackLevel: none/broker, optional (default - broker)
+  accessKey: S3 user access key  
```

Comment:
why do we need accessKey/secretKey here?

Reply:
If we want to be able to talk to the RGW we do, don't we? Unless the operator will directly do radosgw-admin commands. In which case we have to find some way to control what a user is allowed to do like putting notifications only on the buckets he/she is allowed to.

Reply:
the keys are stored in a k8s secret, which is associated with the OBS.
i don't think they should be stored here

Reply:
...actually, topic is not associated to a specific OBC. and can, in theory, be created by a different user, with different credentials.
@travisn how do we manage credentials that are not at bucket level?

Reply:
Rook only manages creating an object user, either with the CephObjectStoreUser CR, or a user that is generated for an OBC for that bucket. The keys are stored in k8s secrets. Here, we should have the name of the secret where the key is loaded from, rather than including the key directly in the CR.

Reply:
@travisn thanks!
So, in both cases (the topic and the notification CRs) we should reference the secret.
In the case of the topic (since its not associated with a single OBC), we should have a separate CephObjectStoreUser CR.